tasks/csv2kdi/csv2kdi.rb

```diff
@@ -274,34 +307,55 @@ def run(opts)
           ### ASSOCIATE THE ASSET TO THE VULN
 
           if @assets_only == "false" # Added for ASSET ONLY Run
-
-            create_asset_vuln(hostname, ip_address, file, mac_address, netbios, url, ec2, fqdn, external_id, database, scanner_type, scanner_id, details, created, scanner_score, last_fixed,
-                              last_seen, status, closed, port)
-
-            # CREATE A VULN DEF THAT HAS THE SAME ID AS OUR VULN
+            if @appsec_findings == "false"
+              create_asset_vuln(hostname, ip_address, file, mac_address, netbios, url, ec2, fqdn, external_id, database, scanner_type, scanner_id, details, created, scanner_score, last_fixed,
+                                last_seen, status, closed, port)
+            else
+              ### ASSOCIATE THE ASSET TO THE findings/vuln
+              create_asset_findings(file, url, external_id, scanner_type, scanner_id, additional_fields, created, scanner_score,
+                                    last_seen, status, closed)
+            end
+            # CREATE A VULN DEF THAT HAS THE SAME ID AS OUR VULN/finding
             create_vuln_def(scanner_type, scanner_id, cve_id, wasc_id, cwe_id, name, description, solution)
           end
-        end
 
-        # puts JSON.pretty_generate kdi_output
+          if kdi_entry_total > 9999
```

Comment:
What happens if the KDI_entry_Total is < 9999?

Reply:
Processing will continue and file will formatted outside the loop.  This was injected to incorporate chunk processing for performance reasons.  Otherwise array and array processing gets to burdensome due to size and memory.

Reply:
Thanks for the clarification, that makes sense.